src/ObjectGraph.es6.js

```diff
@@ -263,6 +293,12 @@ ObjectGraph.prototype.visitObject = function(o) {
     this.q.enqueue(this.visitProperty.bind(this, o, names[i], dataMap));
   }
 
+  // A instance is an object that is not a __proto__ of other object,
```

Comment:
So this strategy isn't sound, right? We can't know whether or not this is an instance until we have visited the entire object graph. Add a `TODO` explaining why this is unsound, please. We can fix this when we refactor this extra logic into a separate component/plugin.